projects/users/src/favorites-service.ts

```diff
@@ -0,0 +1,24 @@
+import { UsersService } from './users-service';
+
+export interface FavoriteUpdateOptions {
+  userId: string;
+  soundId: string;
+}
+
+export class FavoritesService extends UsersService {
+  async addToFavorites(options: FavoriteUpdateOptions): Promise<void> {
+    const collection = await this.usersCollection;
+    await collection.updateOne({ userId: options.userId }, { $push: { favorites: options.soundId } }, { upsert: true });
```

Comment:
Use `$addToSet` rather than `$push`, as it won't allow a duplicate key to be added to the array.

```suggestion
    await collection.updateOne({ userId: options.userId }, { $addToSet: { favorites: options.soundId } }, { upsert: true });
```